compiler/src/iree/compiler/Dialect/HAL/Target/TargetOptions.cpp

```diff
@@ -71,11 +73,17 @@ void TargetOptions::bindOptions(OptionsBinder &binder) {
                      "(- for stdout)."),
       llvm::cl::cat(halTargetOptionsCategory));
 
-  binder.opt<std::string>("iree-hal-dump-executable-intermediates-to",
-                          executableIntermediatesPath,
-                          llvm::cl::desc("Path to write translated executable "
-                                         "intermediates (.bc, .o, etc) into."),
-                          llvm::cl::cat(halTargetOptionsCategory));
+  binder.opt<std::string>(
+      "iree-hal-dump-executable-intermediates-to", executableIntermediatesPath,
+      llvm::cl::desc("Path to write translated executable "
+                     "intermediates (.bc, .o, etc) into."),
+      llvm::cl::cat(halTargetOptionsCategory));
+
+  binder.opt<std::string>(
+      "iree-hal-dump-llvm-ir-to", executableLLVMIRPath,
+      llvm::cl::desc(
+          "Path to write translated and serialized llvmir (.ll) into."),
+      llvm::cl::cat(halTargetOptionsCategory));
```

Comment:
> Useful! LLVM IR is not a universal concept, though, so please remove the new flag and continue to use the old one (as it looks like this anyway?).

+1, I'd just bundle these together and output both/all intermediate files when `iree-hal-dump-executable-intermediates-to` is set

That matches what we do for WebGPU, dumping both the text (human readable) and binary (machine readable) files: https://github.com/iree-org/iree/blob/cc0c8266aa4d457705fcbb94804696ee0c8dcf18/compiler/plugins/target/WebGPUSPIRV/WebGPUSPIRVTarget.cpp#L220-L240

Reply:
@benvanik @ScottTodd Thank you for reviewing. I have updated the commit to bundle both intermediate files together, added this change to CUDATarget, and updated the developer documentation. It seems that ROCM already dumps .ll in the source. As I am not very familiar with other targets, please feel free to let me know if I missed anything else. https://github.com/iree-org/iree/blob/e12e4b4badd38151c67bcf6f32f031cea076c8c6/compiler/plugins/target/ROCM/ROCMTarget.cpp#L474-L486